### PR TITLE
chore: implement Disposable/AsyncDisposable on Page and Browser

### PR DESCRIPTION
**Pull request by jrandolf-google**

### CODE REVIEW

File: packages/puppeteer-core/src/api/Page.ts [hunk 1]
```diff
@@ -2873,6 +2874,14 @@ export class Page extends EventEmitter {
   waitForDevicePrompt(): Promise<DeviceRequestPrompt> {
     throw new Error('Not implemented');
   }
+
+  [Symbol.dispose](): void {
+    return void this.close().catch(debugError);
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/api/Page.ts:2879]: Remove return on Sync dispose.

Comment (jrandolf) [reply packages/puppeteer-core/src/api/Page.ts:2879]: Hmm, but I like the return ;).

Comment (Lightning00Blade) [reply packages/puppeteer-core/src/api/Page.ts:2879]: Why are we returning a promise when it should be void then? Code currently does that at run time.

Comment (jrandolf) [reply packages/puppeteer-core/src/api/Page.ts:2879]: `void promise` returns undefined.

**Review by OrKoN — APPROVED** [review]
LGTM thanks

File: packages/puppeteer-core/src/api/Page.ts (hunk 1 above)
Comment (sadym-chromium) [inline packages/puppeteer-core/src/api/Page.ts:2879]: I'd assume that when the page is disposing, it's already closed, and the only things needed is to clean up the state and remove the listeners. Using `close` here seems inappropriate, as `close` is intended to close the page. I'm sure that re-using should be either:
1. Other way around, and `close` should eventually call `dispose`, or
2. `close` should not call `dispose` at all, as the protocol event listeners will process corresponding event (`browsingContext.contextDestroyed` in vase of BiDi).

**Pull request merged**